### PR TITLE
update document for adding a float compatibility problem's warning

### PR DESCRIPTION
**Pull request by Reminiscent**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add a warning for a float compatibility problem
Related to https://github.com/pingcap/tidb/issues/11328

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Reminiscent** [comment]
@XuHuaiyu PTAL

File: data-type-numeric.md [hunk 1]
```diff
@@ -145,7 +145,8 @@ FLOAT(p) [UNSIGNED] [ZEROFILL]
 
 > **警告：**
 >
-> 与在 MySQL 中一样，`FLOAT` 数据类型存储近似值。对于货币之类的精确值，建议使用 `DECIMAL` 类型。
+> 1. 与在 MySQL 中一样，`FLOAT` 数据类型存储近似值。对于货币之类的精确值，建议使用 `DECIMAL` 类型。
+> 2. 在 TiDB 中，`FLOAT` 数据类型默认的精度是 8 位，与 MySQL 不同。在 MySQL 中，`FLOAT` 数据类型默认的精度是 6 位。
```
Comment (XuHuaiyu) [inline data-type-numeric.md:149]: 举个例子？

Comment (Reminiscent) [reply data-type-numeric.md]: @XuHuaiyu Update. PTAL

**Review by XuHuaiyu — COMMENTED** [review]
LGTM

File: data-type-numeric.md [hunk 2]
```diff
@@ -145,7 +145,8 @@ FLOAT(p) [UNSIGNED] [ZEROFILL]
 
 > **警告：**
 >
-> 与在 MySQL 中一样，`FLOAT` 数据类型存储近似值。对于货币之类的精确值，建议使用 `DECIMAL` 类型。
+> 1. 与在 MySQL 中一样，`FLOAT` 数据类型存储近似值。对于货币之类的精确值，建议使用 `DECIMAL` 类型。
+> 2. 在 TiDB 中，`FLOAT` 数据类型默认的精度是 8 位，与 MySQL 不同。在 MySQL 中，`FLOAT` 数据类型默认的精度是 6 位。举例如下，同时往 MySQL 和 TiDB 中类型为 `FLOAT` 的列中插入数据 123456789 和 1.23456789。在 MySQL 中查询对应的值，将会得到结果 123457000 和 1.23457。而在 TiDB 中查询对应的值，将会得到结果 123456790 和 1.2345679。
```
Comment (TomShawn) [inline data-type-numeric.md:149]:
```suggestion
> + 与在 MySQL 中一样，`FLOAT` 数据类型用于存储近似值。对于类似货币的精确值，建议使用 `DECIMAL` 类型。
> + 在 TiDB 中，`FLOAT` 数据类型默认的精度是 8 位，这与 MySQL 不同。在 MySQL 中，`FLOAT` 数据类型默认的精度是 6 位。例如，同时往 MySQL 和 TiDB 中类型为 `FLOAT` 的列中插入数据 `123456789` 和 `1.23456789`。在 MySQL 中查询对应的值，将会得到结果 `123457000` 和 `1.23457`。而在 TiDB 中查询对应的值，将会得到 `123456790` 和 `1.2345679`。
```

File: data-type-numeric.md [hunk 3]
```diff
@@ -145,7 +145,8 @@ FLOAT(p) [UNSIGNED] [ZEROFILL]
 
 > **警告：**
```
Comment (TomShawn) [inline data-type-numeric.md:146]: 这个注意类型能有`警告`的级别？

Comment (Reminiscent) [reply data-type-numeric.md:146]: @XuHuaiyu 这里用`警告`级别么，还是用其他啥级别的说明呢

Comment (XuHuaiyu) [reply data-type-numeric.md:146]: 这个警告是不是 `warning` 的翻译

感觉新加的这条这个更像一个 `注意事项`的意思?

Comment (TomShawn) [inline data-type-numeric.md:146]:
```suggestion
> **注意：**
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**